src/mult.jl

```diff
@@ -138,3 +151,20 @@ end
 function MA.mutable_operate!(::typeof(*), p::Polynomial{C}, q::Polynomial{C}) where C
     return MA.mutable_operate_to!(p, *, p, q)
 end
+
+# Overwrite this method for monomial-like terms because
+# otherwise it would check `iszero(α)` and in that case 
+# dismiss of the variable of `p` by performing 
+# `operate_to!(zero, output :: Polynomial )` which only
+# respects the variables that are stored already
+function MP._multconstant_to!(output::Polynomial, α, f, p :: DMonomialLike)
+    if iszero(α)
+        empty!(output.a)
+        empty!(output.x.vars)
+        push!(output.x.vars, variables(p)...)
```

Comment:
Use https://github.com/JuliaAlgebra/DynamicPolynomials.jl/blob/9e4d3249d4b3c87df7179c213a67e7750c6e8217/src/monovec.jl#L255
`variables(p)...` will have to compile a new method for each number of variables